### PR TITLE
[playground] Improve experience when the server is not accessible or under heavy load

### PR DESCRIPTION
**Pull request by wkazmierczak**
- [x] Attempt to send request multiple times.
- [x] Set timeout for fetch.

### CODE REVIEW

File: docs/src/api.ts [hunk 1]
```diff
@@ -53,21 +53,64 @@ async function createError(response: Response): Promise<ApiError> {
...
+async function fetchWithTimeout(
+  path: string,
+  requestObject: RequestObject,
+  timeout: number
+): Promise<Response> {
+  const backendUrl: URL = new URL(
     config.customFields.environment === 'development'
       ? 'http://localhost:8081'
       : 'https://playground.compositor.live'
   );
```
Comment (wkozyra95) [inline docs/src/api.ts:68]:
```suggestion
  const backendUrl: string = config.customFields.environment === 'development'
      ? 'http://localhost:8081'
      : 'https://playground.compositor.live';
```

you could also move this value to the global const on top of the file

File: docs/src/api.ts [hunk 2]
```diff
@@ -53,21 +53,64 @@ async function createError(response: Response): Promise<ApiError> {
...
+      })
+      .catch(error => {
+        clearTimeout(timer);
+        rej(new ApiError(error.message));
+      });
+  });
+}
+
+async function retryRequest(func, path, requestObject) {
+  const delay = 1000;
```
Comment (wkozyra95) [inline docs/src/api.ts:93]:
```suggestion
  const DELAY_MS = 1000;
```
use upper case for const, if representing time value add unit to the name
the same applies to values bellow

File: docs/src/api.ts [hunk 3]
```diff
@@ -53,21 +58,59 @@ async function createError(response: Response): Promise<ApiError> {
...
+        res(response);
+      })
+      .catch(error => {
+        clearTimeout(timer);
+        rej(new ApiError(error.message));
+      });
+  });
+}
+
+async function retryRequest(path, requestObject) {
```
Comment (wkazmierczak) [inline docs/src/api.ts:92]: I wasn't sure what exactly should remain in `retryRequest` function, so for now I did it like this

File: docs/src/api.ts [hunk 4]
```diff
@@ -52,22 +57,66 @@ async function createError(response: Response): Promise<ApiError> {
...
-      : 'https://playground.compositor.live'
-  );
-  const renderImageUrl = new URL('/render_image', backend_url);
-  let response;
-  try {
-    response = await fetch(renderImageUrl, requestObject);
-  } catch (error) {
-    throw new ApiError(error.message);
-  }
+  const response = await retryWithSleep(fn);
```
Comment (wkozyra95) [inline docs/src/api.ts:64]:
```suggestion
  const response = await retryWithTimeout(async () => (
    await fetchWithTimeout('/render_image', requestObject, TIMEOUT_MS)
  ), TIMEOUT_MS);
```

File: docs/src/api.ts [hunk 5]
```diff
@@ -52,22 +57,66 @@ async function createError(response: Response): Promise<ApiError> {
...
+        res(response);
+      })
+      .catch(error => {
+        clearTimeout(timer);
+        rej(new ApiError(error.message));
+      });
+  });
+}
+
+async function retryWithSleep<T>(fn: () => Promise<T>): Promise<T> {
```
Comment (wkozyra95) [inline docs/src/api.ts:99]:
```suggestion
async function retryWithSleep<T>(fn: () => Promise<T>, timeoutMs: number): Promise<T> {
```

**Pull request merged**